test/test.ts

```diff
@@ -110,7 +110,7 @@ describe('linkinator', () => {
       path: 'test/fixtures/recurse',
       recurse: true,
     });
-    assert.strictEqual(results.links.length, 5);
+    assert.strictEqual(results.links.length, 4);
```

Comment:
Turns out this was a bug!  The cache was not normalizing the url for `/index` style paths, which the new `URL` object does 😄